### PR TITLE
fix keyerror in config.rename

### LINKED ISSUE #4902: keyerror in dask.config.rename when working with distributed
calling `dask-worker` using `dask==1.2.2`, `distributed==1.28.1`, I get the following error:

`Traceback (most recent call last): File "/opt/conda/envs/worker/bin/dask-worker", line 7, in <module> from distributed.cli.dask_worker import go File "/opt/conda/envs/worker/lib/python3.6/site-packages/distributed/__init__.py", line 3, in <module> from . import config File "/opt/conda/envs/worker/lib/python3.6/site-packages/distributed/config.py", line 55, in <module> dask.config.rename(aliases) File "/opt/conda/envs/worker/lib/python3.6/site-packages/dask/config.py", line 451, in rename del config[k] # TODO: support nested keys KeyError: 'tick-maximum-delay'`

It looks like this was introduced by #4742 when `dask.config.rename` was not properly adjusted, so that it can run into these KeyErrors. Current state of the function:

```def rename(aliases, config=config):
    """ Rename old keys to new keys
    This helps migrate older configuration versions over time
    """
    old = []
    new = {}
    for o, n in aliases.items():
        value = get(o, None, config=config)
        if value is not None:
            old.append(o)
            new[n] = value

    for k in old:
        del config[k]  # TODO: support nested keys

    set(new, config=config)
```

Suggest changing the 3rd to last line to `del config[canonical_name(k, config)]`. Will file an appropriate PR.

### PR DESCRIPTION
**Pull request by bolliger32**
Fixes #4902 

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
Thanks for the fix @bolliger32 !  

Do you have any thoughts on what a small test would be for this change?  It would be good to have such a test to make sure that your work doesn't get reverted in the future by some careless change.

**Comment by bolliger32** [comment]
Ah good call! Will think something up and push it. Thanks for working w/ a first-timer.

**Comment by bolliger32** [comment]
ok @mrocklin just needed a tiny change to an existing test and I think we're good.

File: dask/config.py [hunk 1]
```diff
@@ -448,7 +448,7 @@ def rename(aliases, config=config):
             new[n] = value
 
     for k in old:
-        del config[k]  # TODO: support nested keys
+        del config[canonical_name(k,config)]  # TODO: support nested keys
```
Comment (mrocklin) [inline dask/config.py]: I suspect also that the linter will yell at you here.  It'll want a space after the comma.

We use flake8 to lint our code.  You can check this locally by running the following from the root directory

```
pip install flake8
flake8 dask
```

Comment (bolliger32) [reply dask/config.py]: ah cool I was just getting to installing flake8 and figuring out what it was. Have added the space and run it and it passes (though it didn't seem to raise any issues without the space either)

**Comment by mrocklin** [comment]
This is in.  Thanks @bolliger32 

> Thanks for working w/ a first-timer.

Could have fooled me :)  Welcome to the project.

**Pull request merged**